nolearn/lasagne/base.py

```diff
@@ -256,8 +268,12 @@ def _create_iter_funcs(self, layers, objective, update, input_type,
         update_params = self._get_params_for('update')
         updates = update(loss_train, all_params, **update_params)
 
-        X_inputs = [theano.Param(X_batch)]
-        inputs = X_inputs + [theano.Param(y_batch)]
+        input_layers = [layer for layer in layers.values()
+                        if isinstance(layer, InputLayer)]
+
+        X_inputs = [theano.Param(input_layer.input_var, name=input_layer.name)
```

Comment:
Fair point.

On Fri, May 29, 2015 at 6:48 PM Daniel Nouri <notifications@github.com>
wrote:

> In nolearn/lasagne/base.py
> <https://github.com/dnouri/nolearn/pull/106#discussion_r31371435>:
>
> > @@ -256,8 +268,12 @@ def _create_iter_funcs(self, layers, objective, update, input_type,
> >          update_params = self._get_params_for('update')
> >          updates = update(loss_train, all_params, **update_params)
> >
> > -        X_inputs = [theano.Param(X_batch)]
> > -        inputs = X_inputs + [theano.Param(y_batch)]
> > +        input_layers = [layer for layer in layers.values()
> > +                        if isinstance(layer, InputLayer)]
> > +
> > +        X_inputs = [theano.Param(input_layer.input_var, name=input_layer.name)
>
> The reason I changed it was because it seemed nicer than writing fit({'layer_name.X':
> X}, y).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dnouri/nolearn/pull/106/files#r31371435>.
>